pulsar/reader_partition.go

```diff
@@ -0,0 +1,142 @@
+package pulsar
+
+import (
+	"sync"
+	"time"
+)
+
+func (r *reader) internalTopicReadToPartitions() error {
```

Comment:
What is this function doing I'm a little confused by the name. Also, this looks like a lot of duplication code from the consumer partitions is there any way to refactor/combine in to helper functions to avoid duplication?

Reply:
According to issues, we need to subscribe to partition_consumer separately. The meaning of this function is to convert topic into partition_topic for subscription

Reply:
What's more difficult is that reader requires a lot of specific parameters. After thinking about it, I decided to encapsulate directly. If not, this will swell the member variables of the reader.

Reply:
For example, the Reader will process startMessageID, but the consumer will not.